src/south/sonic-interface/main.cpp

```diff
@@ -1,129 +1,463 @@
+* Description   : Callback for curl fetch                                    *
+* Input         : void *contents                                             *
+*                 size_t size                                                *
+*                 size_t nmemb                                               *
+*                 void *userp                                                *
+* Output        : size_t                                                     *
+*****************************************************************************/
+size_t
+curl_callback (void *contents, size_t size, size_t nmemb, void *userp)
+{
+  size_t realsize = size * nmemb;
+  struct curl_fetch_st *p = (struct curl_fetch_st *) userp;
+  p->payload = (char *) realloc (p->payload, p->size + realsize + 1);
+
+  /* check buffer */
+  if (p->payload == NULL)
+    {
+      fprintf (stderr, "ERROR: Failed to expand buffer in curl_callback");
+      /* free buffer */
+      free (p->payload);
+      /* return */
+      return 1;
+    }
+
+  /* copy contents to buffer */
+  memcpy (&(p->payload[p->size]), contents, realsize);
+  /* set new buffer size */
+  p->size += realsize;
+  /* ensure null termination */
+  p->payload[p->size] = 0;
+  /* return size */
+  return realsize;
+}
+
+/*****************************************************************************
+* Function Name : oper_get_items                                             *
+* Description   : To get the Operational data                                *
+* Input         : sysrepo::S_Session session                                 *
+*                 const char *module_name                                    *
+*                 const char *path                                           *
+*                 const char *request_xpath                                  *
+*                 uint32_t request_id                                        *
+*                 libyang::S_Data_Node &parent                               *
+*                 void *private_data                                         *
+* Output        : int                                                        *
+*****************************************************************************/
+int
+SonicController::oper_get_items (sysrepo::S_Session session,
+                                 const char *module_name,
+                                 const char *path,
+                                 const char *request_xpath,
+                                 uint32_t request_id,
+                                 libyang::S_Data_Node &parent,
+                                 void *private_data)
+{
+  std::stringstream url;
+  libyang::S_Context ly_ctx = session->get_context();
+
+  CURL *curl;
+  CURLcode res;
+  struct curl_fetch_st curl_fetch;
+  struct curl_fetch_st *fetch = &curl_fetch;
+  fetch->payload = (char *) calloc (1, sizeof(fetch->payload));
+  char *payload = (char *) calloc (1, MAX_PAYLOAD_SIZE);
```

Comment:
why needed?

Reply:
We may be able to remove this as well

https://github.com/microsonic/goldstone-mgmt/pull/23/files#diff-f5a830eeff1bf48f8cc57fdebfd8cd82R20

Reply:
Done, Removed payload